### PR TITLE
Merge branch 1.6 back to master

### PR DESCRIPTION
**Pull request by gunan**

### CODE REVIEW

File: tensorflow/contrib/tpu/python/tpu/tpu_estimator.py [hunk 1]
```diff
@@ -1660,6 +1660,16 @@ def __init__(self,
         util_lib.check_positive_integer(predict_batch_size,
                                         'predict_batch_size')
 
+      if predict_batch_size is not None:
```
Comment (jhseu) [inline tensorflow/contrib/tpu/python/tpu/tpu_estimator.py]: Revert this file?

File: tensorflow/docs_src/install/install_linux.md [hunk 2]
```diff
@@ -293,7 +293,11 @@ take the following steps:
 
      <pre>
      $ <b>sudo pip3 install --upgrade \
+<<<<<<< HEAD
```
Comment (jhseu) [inline tensorflow/docs_src/install/install_linux.md]: Merge conflicts in this file

File: tensorflow/contrib/cmake/external/protobuf.cmake [hunk 3]
```diff
@@ -16,7 +16,7 @@ include (ExternalProject)
 
 set(PROTOBUF_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/protobuf/src/protobuf/src)
 set(PROTOBUF_URL https://github.com/google/protobuf.git)
-set(PROTOBUF_TAG 396336eb961b75f03b25824fe86cf6490fb75e3a)
+set(PROTOBUF_TAG b04e5cba356212e4e8c66c61bbe0c3a20537c5b9)
```
Comment (av8ramit) [inline tensorflow/contrib/cmake/external/protobuf.cmake]: I don't think we want to keep this change.

File: tensorflow/core/distributed_runtime/tensor_coding.cc [hunk 4]
```diff
@@ -81,7 +81,7 @@ void TensorResponse::InitPartial(const RecvTensorResponse& response) {
 Status TensorResponse::ParseFrom(Source* source) {
   if (!on_host_) {
     protobuf::io::CodedInputStream input(source->contents());
-    input.SetTotalBytesLimit(INT_MAX);  // Unlimited
+    input.SetTotalBytesLimit(INT_MAX, INT_MAX);  // Unlimited
```
Comment (av8ramit) [inline tensorflow/core/distributed_runtime/tensor_coding.cc]: We don't want this change.

File: tensorflow/python/training/saver.py [hunk 5]
```diff
@@ -1597,9 +1597,9 @@ def save(self,
         [Stripping Default-Valued Attributes](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md#stripping-default-valued-attributes).
 
     Returns:
-      A string: path prefix used for the checkpoint files. If checkpoint
-        format is V1 and the saver is sharded, this string ends with:
-         '-?????-of-nnnnn' where 'nnnnn' is the number of shards created.
+      A string: path prefix used for the checkpoint files.  If the saver is
+        sharded, this string ends with: '-?????-of-nnnnn' where 'nnnnn'
+        is the number of shards created.
```
Comment (av8ramit) [inline tensorflow/python/training/saver.py]: Revert this file

File: tensorflow/tools/ci_build/windows/gpu/cmake/run_build.bat [hunk 6]
```diff
@@ -37,7 +37,7 @@ SET CMAKE_DIR=%REPO_ROOT%\tensorflow\contrib\cmake
 SET MSBUILD_EXE="C:\Program Files (x86)\MSBuild\14.0\Bin\msbuild.exe"
 
 :: Run cmake to create Visual Studio Project files.
-%CMAKE_EXE% %CMAKE_DIR% -A x64 -DSWIG_EXECUTABLE=%SWIG_EXE% -DPYTHON_EXECUTABLE=%PY_EXE% -DCMAKE_BUILD_TYPE=Release -DPYTHON_LIBRARIES=%PY_LIB% -Dtensorflow_BUILD_PYTHON_TESTS=%BUILD_PYTHON_TESTS% -Dtensorflow_BUILD_CC_TESTS=%BUILD_CC_TESTS% -Dtensorflow_ENABLE_GPU=ON -DCUDNN_HOME=%CUDNN_HOME% -Dtensorflow_TF_NIGHTLY=%TF_NIGHTLY% -Dtensorflow_DISABLE_EIGEN_FORCEINLINE=%DISABLE_FORCEINLINE% -Dtensorflow_WIN_CPU_SIMD_OPTIONS=/arch:AVX
```
Comment (av8ramit) [inline tensorflow/tools/ci_build/windows/gpu/cmake/run_build.bat]: Revert this file for now. We didn't prove this works on kokoro. We can fix it cleanly in master.

**Review by av8ramit — CHANGES_REQUESTED** [review]
Revert a majority of files. Only thing I think should come through are version changes and the setup.py changes.

**Comment by gunan** [comment]
We do want some of the changes, for example saver.py revert breaks checkpoint loading code, and it does not seem to be in master.

Anyway, how did these branches diverge this much? I did not even touch most of these files myself, I just merged the branch.

**Comment by gunan** [comment]
OK, retrying, PTAL.

File: tensorflow/python/training/saver.py (hunk 5 above)
Comment (gunan) [reply tensorflow/python/training/saver.py]: This is actually needed. We need to roll this back and propagate this rollback also internally to fix the last pull from github.
This file and evaluation_test.py changes are a part of the rollback, I manually verified those two.

Comment (av8ramit) [reply tensorflow/python/training/saver.py]: Okay sounds good.

**Review by av8ramit — APPROVED** [review]

File: tensorflow/contrib/slim/python/slim/evaluation_test.py [hunk 7]
```diff
@@ -235,7 +235,7 @@ def testErrorRaisedIfCheckpointDoesntExist(self):
   def _prepareCheckpoint(self, checkpoint_path):
     init_op = control_flow_ops.group(variables.global_variables_initializer(),
                                      variables.local_variables_initializer())
-    saver = saver_lib.Saver()
+    saver = saver_lib.Saver(write_version=saver_pb2.SaverDef.V1)
```
Comment (av8ramit) [inline tensorflow/contrib/slim/python/slim/evaluation_test.py]: This might be the reason slim evaluation test is failing.

Comment (gunan) [reply tensorflow/contrib/slim/python/slim/evaluation_test.py]: @drpngx looks like I am not able to cleanly rollback saver changes on master.
Has there been more changes I should be rolling back, too?

**Review by jhseu — APPROVED** [review]

File: tensorflow/contrib/slim/python/slim/evaluation_test.py (hunk 7 above)
Comment (drpngx) [reply tensorflow/contrib/slim/python/slim/evaluation_test.py]: I thought it was just https://github.com/tensorflow/tensorflow/pull/14341 and you have these changes.

**Pull request merged**